docs(g11n): add some notes (#1723)
carbon-design-system/carbon: docs/g11n.md

# Notes of globalization

> * `carbon-components` let you define most of the markups, where most of the translatable contents are in. Application needs to make sure such contents in the markups are translatable as needed.

Reviewer: ```suggestion
* `carbon-components` lets you define most of the markup, which contains most of the translatable content. Ensure application content in the markup is translatable as needed.
```

> * Also, from the same reason above, application needs to make sure list items in the markup are in the correct order.

Reviewer: ```suggestion
* Also make sure that list items in the application markup are correctly ordered for the same reason.
```
